java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -59,9 +66,17 @@ public JDBCSink(JDBCSinkConfig config, TableSchema tableSchema) {
         this.config = config;
         try {
             this.conn = DriverManager.getConnection(config.getJdbcUrl());
-            this.conn.setAutoCommit(false);
             this.pkColumnNames =
                     getPkColumnNames(conn, config.getTableName(), config.getSchemaName());
+            // disable auto commit can improve performance
+            this.conn.setAutoCommit(false);
+            // use the lowest isolation level since we don't guarantee exactly-once
+            this.conn.setTransactionIsolation(Connection.TRANSACTION_READ_UNCOMMITTED);
```

Comment:
Auto commit would lead to a long transaction in the downstream database which I think most of our users don't expect. How much performance we can achieve if disable auto-commit? If we use batch execution, there should be no noticeable performance improvement achieved by disabling auto-commit I think. BTW, I think we should use the default transaction isolation level instead of read uncommitted because reading uncommitted data seems a little bit weird to me and we don't guarantee exactly-once doesn't mean we want to see uncommitted data.

Reply:
The test shows that the performance gain with auto commit disabled is huge, while the isolation level makes no much difference to the performance. And I think use read uncommitted in sink is ok, since it just blindly write to the database and doesn't read any data.

<img width="762" alt="Aggregated Sink Throughput(rowss)" src="https://github.com/risingwavelabs/risingwave/assets/7338536/076d09bf-8cf6-46c0-8bf4-a7ec2f03ff3c">

Reply:
Ok, the improvement is great. Can we commit in the `write` method instead of each barrier?

Reply:
@StrikeW Can you also compare the throughput when we disable auto_commit but call commit at each stream chunk instead of at each checkpoint barrier?

Reply:
Sounds ok to me, I'll have a try and test.

Reply:
Seems the throughout is roughly same in both settings. We can commit on for each chunk to improve freshness.
<img width="848" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/7338536/1bd74f34-48c8-4ba8-b697-edcd14d6bf1b">

Reply:
> reading uncommitted data seems a little bit weird to me and we don't guarantee exactly-once doesn't mean we want to see uncommitted data

+1

Reply:
The transaction here is write-only, and the isolation level change would not affect downstream transactions. Any concern for the change?👀

Reply:
`Delete` is somehow a read operation as well, because it need to lookup the table to find those rows matched the pk.

Reply:
> The transaction here is write-only, and the isolation level change would not affect downstream transactions. Any concern for the change?👀

I think it depends on the database implementation. For DB like MySQL and SQLSever, since we are doing a point update/delete on PK here, I think read uncommitted and read committed (the JDBC default) are the same. However, I think it is still better to keep the default given that we believe setting isolation level to read uncommitted is an noop in our use case.

Reply:
The motivation to lower the isolation level is to avoid acquire **read locks** in downstream, since sink doesn't need to read, so no need to worry about read consistency. The `DELETE/UPDATE` command would block if there is any downstream transaction is writing the row, and will proceed after the downstream transaction finished.

Btw, the isolation level doesn't affect much of the sink performance according to the test result, so I am ok to retain it to RC level. So that we don't need to fix https://github.com/risingwavelabs/risingwave/issues/12089 cc @fuyufjh